cms/users/signal_handlers.py

```diff
@@ -0,0 +1,53 @@
+import logging
+from typing import Any
+
+from django.contrib.auth.signals import user_logged_in, user_logged_out, user_login_failed
+from django.dispatch import receiver
+from django.http import HttpRequest
+
+from cms.core.utils import get_client_ip
+
+from .models import User
+
+logger = logging.getLogger("cms.users")
+
+
+@receiver(user_logged_in)
+def audit_user_logged_in(sender: Any, request: HttpRequest, user: User, **kwargs: Any) -> None:  # pylint: disable=unused-argument
+    logger.info(
+        "User logged in",
+        extra={
+            "user_id": user.id,
+            "email": user.email,
+            "event": "user_logged_in",
+            "ip_address": get_client_ip(request),
+            "user_agent": request.headers.get("User-Agent"),
+        },
+    )
+
+
+@receiver(user_logged_out)
+def audit_user_logged_out(sender: Any, request: HttpRequest, user: User, **kwargs: Any) -> None:  # pylint: disable=unused-argument
```

Comment:
`user` can be `None` here if the user was not authenticated before logging out (in other words, this signal is still sent if a user attempted a logout without first logging in). This will throw an exception in the code below.

You can verify this by writing a test which attempts a logout without a prior login. Perhaps we could also add a test to cover this?

https://docs.djangoproject.com/en/5.1/ref/contrib/auth/#django.contrib.auth.signals.user_logged_out

Reply:
Great catch, I missed that. I'll fix the types and add a test!